### PR TITLE
extend ansible bats test to also test role import and execution

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
I don't like the `sleep`s, so it's draft for now

**Comment by ekohl** [comment]
How do you expect to solve the `sleep`? Could hammer be enhanced to have a `--wait/--no-wait` parameter to poll the job? Is there an issue tracking that?

**Comment by evgeni** [comment]
Unsure yet. The second one actually returns a job invocation id, so I can loop on `hammer job-invocation info` until that one is completed. The import… :woman_shrugging: 

Yes, other hammer actions sometimes have `--sync/--async` params, so that might also be a path.

File: bats/fb-test-foreman-ansible.bats [hunk 1]
```diff
@@ -2,7 +2,40 @@
...
+  hammer ansible roles list | grep "${ROLE_NAME}"
+}
+
+@test "run a role" {
+  # assign the role to our system and run it
+  hammer host ansible-roles assign --name $HOSTNAME --ansible-roles "${ROLE_NAME}" | grep 'Ansible roles were assigned to the host'
+  hammer host ansible-roles play --name $HOSTNAME | grep 'Ansible roles are being played.'
+
+  # wait for the run to actually succeed
+  tWaitForTask Actions::RemoteExecution::RunHostsJob
```
Comment (evgeni) [inline bats/fb-test-foreman-ansible.bats:37]: Opened https://projects.theforeman.org/issues/35881 for this.

File: bats/fb-test-foreman-ansible.bats [hunk 2]
```diff
@@ -2,7 +2,40 @@
...
+  # create a dummy role
+  mkdir -p "/etc/ansible/roles/${ROLE_NAME}/tasks"
+  echo "- command: uptime" > "/etc/ansible/roles/${ROLE_NAME}/tasks/main.yml"
+
+  # check if smart_proxy_ansible finds it and can import it
+  hammer ansible roles fetch --proxy-id 1 | grep "${ROLE_NAME}"
+  hammer ansible roles sync --proxy-id 1 --role-names "${ROLE_NAME}"
+
+  # sync creates a task, so let's wait for it to complete
+  tWaitForTask SyncRolesAndVariables
```
Comment (evgeni) [inline bats/fb-test-foreman-ansible.bats:26]: https://projects.theforeman.org/issues/35882

**Review by ehelms — APPROVED** [review]

**Pull request merged**